src/evm.rs

```diff
@@ -109,11 +128,90 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> EvmNamespaceT
             }
         })
     }
+
+    fn evm_mine(&self) -> BoxFuture<Result<String>> {
+        let inner = Arc::clone(&self.node);
+        Box::pin(async move {
+            match inner.write() {
+                Ok(mut inner) => {
+                    let (keys, block, bytecodes) = {
```

Comment:
Should we move out a lot of this code into `utils.rs`, as it will be used again by `hardhat_mine` in https://github.com/matter-labs/era-test-node/issues/75 ?

Reply:
yeah, was considering that although I think for `hardhat_mine` we need `BlockContextMode::OverrideCurrent` instead of `BlockContextMode::NewBlock` (to skip ahead n blocks in constant time). perhaps we should use that for this method too?

Reply:
done, added `hardhat_mine` to this PR